traits/ctraits.c

```diff
@@ -1815,9 +1815,14 @@ default_value_for(trait_object *trait, has_traits_object *obj, PyObject *name)
             result = PyObject_Call(trait->default_value, tuple, NULL);
             Py_DECREF(tuple);
             if ((result != NULL) && (trait->validate != NULL)) {
-                value = trait->validate(trait, obj, name, result);
-                Py_DECREF(result);
-                return value;
+                if (trait->flags & TRAIT_SETATTR_ORIGINAL_VALUE){
```

Comment:
Spacing / style nitpick:
```suggestion
                if (trait->flags & TRAIT_SETATTR_ORIGINAL_VALUE) {
```